docs/device-emulation.md

```diff
@@ -30,33 +30,74 @@ Because we do not want to modify the daemon, we think it makes sense to *load* a
 
-For instance, calling `fwupdmgr emulation-tag` would ask the end user to choose a device to start
+## Getting started
+
+The very first thing to do is to turn on the emulation functionality for fwupd.  It's off by default.
+
+    fwupdmgr modify-config fwupd AllowEmulation true
+
+Restart the daemon.
+
+**NOTE:** *If you are running in the fwupd development environment, you will need to manually restart the
+daemon because dbus activation doesn't work in the fwupd development environment.*
+
+## Tell the daemon to record a hotplug device
+
+If the device you will be recording from is *hotpluggable* you will use the **emulation-tag** feature for the daemon.
+Calling `fwupdmgr emulation-tag` would ask the end user to choose a device to start
 *recording* so that subsequent re-plugs are available to save.
 As the device state may not be persistent we save the device-should-be-recorded metadata in the
 pending database like we would do for a successful firmware update.
 
-To demo this, something like this could be done:
+Here is how a hotplugged device would be recorded from:
 
-    # connect ColorHug2
-    fwupdmgr modify-config AllowEmulation true
+    # connect ColorHug2 and use the device ID to tag it
     fwupdmgr emulation-tag b0a78eb71f4eeea7df8fb114522556ba8ce22074
     # or, using the GUID
     # fwupdmgr emulation-tag 2082b5e0-7a64-478a-b1b2-e3404fab6dad
     # remove and re-insert ColorHug2
     fwupdmgr get-devices --filter emulation-tag
+
+## Tell the daemon to record a persistent device
+
+Persistent devices are coldplugged or physically not removable from the system.  In this case you must
+tell the daemon by a list of device IDs to start recording data immediately at startup by specifying in the config file.
+
+    fwupdmgr modify-config fwupd EmulatedDevices b0a78eb71f4eeea7df8fb114522556ba8ce22074
+
+Restart the daemon.
+
+**NOTE:** *If you are running in the fwupd development environment, you will need to manually restart the
+daemon because dbus activation doesn't work in the fwupd development environment.*
+
+## Record some data
+
+In both cases the flow to record the data is identical. You will use the device as normally and when you're
+done with the update you will record all events into a zip file.
+
     fwupdmgr download https://fwupd.org/downloads/170f2c19f17b7819644d3fcc7617621cc3350a04-hughski-colorhug2-2.0.6.cab
-    fwupdmgr install e5* --allow-reinstall
+    fwupdmgr install 17*.cab --allow-reinstall
     fwupdmgr emulation-save colorhug.zip
-    # remove ColorHug2
+
+## Test your data
+
+Now that you have data recorded you can remove your device from the system and then try to load the emulation
+data.  You should be able to see the emulated device as well as interact with it.
+
     fwupdmgr emulation-load colorhug.zip
     fwupdmgr get-devices --filter emulated
-    fwupdmgr install e5* --allow-reinstall
-    fwupdmgr modify-config AllowEmulation false
+    fwupdmgr install 17*.cab --allow-reinstall
```

Comment:
I was pondering adding this as a wizard to gnome-firmware, it's kinda clunky doing it on the CLI...

Reply:
I posted to the wrong PR, but I also asked for clarification about the argument `e5* `. I was also wondering if emulation capture is possible with fwupdtool and local cab files?

Reply:
Yeah you can do it with local cab files. The reason for using LVFS ones is to build a reproducible device test.

Reply:
> I was pondering adding this as a wizard to gnome-firmware, it's kinda clunky doing it on the CLI...

In the same vein - should `AllowEmulation` default to enabled?

Reply:
>  was also wondering if emulation capture is possible with fwupdtool 

Realized I forgot to answer this one.  AFAICT It is technically possible now from daemon perspective due to `EmulatedDevices` but we don't have matching code in `fu-tool.c`.  What's the usecase for doing it from `fwupdtool` instead of `fwupdmgr`?

> In the same vein - should AllowEmulation default to enabled?

https://github.com/fwupd/fwupd/pull/7847

Reply:
> Realized I forgot to answer this one. AFAICT It is technically possible now from daemon perspective due to `EmulatedDevices` but we don't have matching code in `fu-tool.c`. What's the usecase for doing it from `fwupdtool` instead of `fwupdmgr`?

No worries, I was under the assumption that fwupdtool was meant for developers and that emulation capture had the same use case when doing local testing.